compute/cloud-client/src/test/java/compute/reservation/ConsumeReservationIT.java

```diff
@@ -0,0 +1,118 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compute.reservation;
+
+import static com.google.cloud.compute.v1.ReservationAffinity.ConsumeReservationType.NO_RESERVATION;
+import static com.google.common.truth.Truth.assertWithMessage;
+
+import com.google.cloud.compute.v1.Instance;
+import com.google.cloud.compute.v1.InstanceTemplate;
+import com.google.cloud.compute.v1.InstanceTemplatesClient;
+import com.google.cloud.compute.v1.InstancesClient;
+import compute.DeleteInstance;
+import compute.DeleteInstanceTemplate;
+import compute.Util;
+import java.io.IOException;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+@Timeout(value = 25, unit = TimeUnit.MINUTES)
+public class ConsumeReservationIT {
+
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String ZONE = "us-central1-a";
+  private static InstancesClient instancesClient;
+  private static InstanceTemplatesClient instanceTemplatesClient;
+  static String javaVersion = System.getProperty("java.version").substring(0, 2);
+  private static final String INSTANCE_NOT_CONSUME_RESERVATION_NAME =
+      "test-instance-not-consume-"  + javaVersion  + "-"
+          + UUID.randomUUID().toString().substring(0, 8);
+  private static final String TEMPLATE_NOT_CONSUME_RESERVATION_NAME =
+      "test-template-not-consume-"  + javaVersion  + "-"
+      + UUID.randomUUID().toString().substring(0, 8);
+
+  // Check if the required environment variables are set.
+  public static void requireEnvVar(String envVarName) {
+    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+        .that(System.getenv(envVarName)).isNotEmpty();
+  }
+
+  @BeforeAll
+  public static void setUp()
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+
+    // Initialize clients once for all tests
+    instancesClient = InstancesClient.create();
+    instanceTemplatesClient = InstanceTemplatesClient.create();
+
+    // Cleanup existing stale resources.
+    Util.cleanUpExistingInstances("test-instance-not-consume-"  + javaVersion, PROJECT_ID, ZONE);
+    Util.cleanUpExistingInstanceTemplates("test-template-not-consume-"  + javaVersion, PROJECT_ID);
```

Comment:
clean up code should not reside in setup method. the test should clean up _after_ it is completed. leaving resources behind unnecessary increases both consumption and billing costs.

Reply:
I think we might want to try and rethink how we do the whole setup-cleanup dance.

I think only Java samples started with the "clean everything" approach years ago and we kept doing it this way, but perhaps it's time to move to an approach similar to other languages - cleanup only what was created in current session. But that would definitely need to be different PR.

Reply:
@m-strzelczyk , should I remove cleanUpExistingInstances() and cleanUpExistingInstanceTemplates() methods from my code in this PR?

Reply:
Please, consult with JUnit guidelines about what should be placed in setup and in cleanup methods. The rule of thumb is the setup performs initialization, including environment provisioning and setup and cleanup restores the environment to what it was before the setup was called and the all the tests in this class were executed.

Reply:
Deleted cleanup methods